app/backend/src/couchers/languages.py

```diff
@@ -0,0 +1,19 @@
+import functools
+import json
+from pathlib import Path
+
+
+@functools.cache
+def get_language_list():
+    """
+    Get list of allowed languages as a set.
+    """
+    with open(Path(__file__).parent / "constant_data" / "languages.json", "r") as f:
+        return set(language["code"] for language in json.loads(f.read()))
+
+
+def check_language_allowed(code):
+    """
+    Check a language code is valid
+    """
```

Comment:
I usually tend to use "check" for functions that raise exception. For functions returning bool, it is better to name it `language_is_allowed`. One of the hard problems in computer science :D